### DISCUSSION TITLE
How do I activate Hot Reload from CLI?

### DISCUSSION THREAD

**Discussion by marinasundstrom** [dotnet/maui#3111] [Q&A] [answered]
How do I tell the CLI to activate Hot Reload for .NET mobile apps?

The commands given in the [docs](https://docs.microsoft.com/en-us/dotnet/maui/get-started/first-app?pivots=dotnet-cli). But no option for Debugging or activating Hot Reload.

```
dotnet build -t:Run -f net6.0-android
dotnet build -t:Run -f net6.0-ios
dotnet build -t:Run -f net6.0-maccatalyst
```

**Comment by gpproton** [comment]
It's really is frustrating with no option

**Reply by marinasundstrom** [reply]
I have not got any answer from anyone about this - even before the "Hot Reload" incident.
I would assume that Watch and Hot Reload (iOS and Android) would work out of the box with no special IDE support.

**Comment by marinasundstrom** [comment, marked as answer]
@LyalinDotCom [answered](https://twitter.com/LyalinDotCom/status/1453051819995779075) on Twitter:

```
Hi! I’m the PM for Hot Reload.  Support for MAUI hot reload is only through the Visual Studio debugger in .NET 6. 
MAUI has a lot of special considerations technically so we made a decision early to get debugger fully working first.
```

So the answer is that there is not option for this yet.

**Reply by marinasundstrom** [reply]
Essentially, it is tied to the proprietary VS Debugger. I now remember them talking about "Hot Reload" being a continuation of "Edit & Continue".

**Reply by matt-goldman** [reply]
Would be great if we could get some guidance as to whether this is in the roadmap.

**Comment by nor0x** [comment]
any updates on this 👀
@davidortinau is hot reload via CLI going to be in the GA version of MAUI?

**Comment by Zee2** [comment]
+1, I've been trying to use the default MAUI template to build and run an app on Windows, and it currently straight up doesn't work from CLI (see https://github.com/dotnet/maui/issues/4329). Apparently `dotnet build` to Windows just doesn't work... and needing to use Visual Studio for this is frustrating!

**Reply by xiangnanscu** [reply]
is it possible to hot reload on macOS?

**Reply by alfarahn** [reply]
@Zee2 @nor0x @matt-goldman @marinasundstrom hey people I work on the customer side but I got this to work you can check:
https://www.linkedin.com/feed/update/urn:li:activity:6987367464254717952/

This is what I did from high level:
1. Created a codespace that is Linux and have nested virtualization capabilities
2. Installed android sdk
3. Installed android cmdlinetools
4. Installed maui android workload
5. run a docker emulator for android
6. connect adb to the emulator
7. build and run the app on ctrl+s

The tooling needs maturity, but that's how I got everything to work in my video.
